packages/components/src/Bottega.purs

```diff
@@ -30,12 +32,18 @@ getOrder { userId, authToken } orderNumber = do
     authorization = oauthToken authToken
     authUser = unsafeToForeign userId
 
-readOrder :: { number :: OrderNumber, user :: UUID, status :: { state :: Foreign, time :: String } } -> Aff Order
+readOrder :: { number :: OrderNumber, user :: UUID, status :: { state :: Foreign, time :: String, failReason :: Nullable String } } -> Aff Order
 readOrder orderObj = do
   orderStatus <- case read orderObj.status.state of
     Right status -> pure status
     Left err     -> pure UnknownState
-  pure $ orderObj { status { state = orderStatus } }
+  failureReason <- case orderStatus of
+    UnknownState -> pure Nothing
+    _ -> pure $ do
+      reasonString <- toMaybe orderObj.status.failReason
+      reason       <- parseFailReason reasonString
+      pure reason
+  pure $ orderObj { status { state = orderStatus, failReason = failureReason } }
```

Comment:
@vapaj in this version `orderObj.status.failReason` is a `Nullable String`.

The function `parseFailReason` has signature `String -> Maybe OrderStatusFailReason`.

Is this what you meant? Can I refactor the `status` field in a similar fashion (except it's not `Nullable`)?

Reply:
> The function `parseFailReason` has signature `String -> Maybe OrderStatusFailReason`

I think it does not have to be a `Maybe`, as we have the `UnknownReason` constructor in case we cannot parse it (so `String -> OrderStatusFailReason`).

> Can I refactor the `status` field in a similar fashion (except it's not `Nullable`)

Yep 👍

Reply:
> I think it does not have to be a Maybe

That was done like this so I could use monadic style to propagate the possible `Nothing` coming from the `toMaybe` call without explicitly handling it

Reply:
Because `Maybe` implements `Functor`, you can use `map` here to achieve this:
```
reason <- map parseFailReason $ toMaybe orderObj.status.failReason
```
(https://pursuit.purescript.org/packages/purescript-maybe/4.0.1/docs/Data.Maybe#t:Maybe)